opencog/query/PatternMatchEngine.cc

```diff
@@ -693,6 +693,17 @@ bool PatternMatchEngine::glob_compare(const PatternTermSeq& osp,
 	GlobGrd glob_grd;
 	GlobPosStack glob_pos_stack;
 
+	// A variable or glob in osp may have been grounded before
+	// reaching here if it exists in some other clause also.
+	// Clear only those that are grounded here in glob_compare().
+	// solution_push/pop is not used as it seems to become a bit
+	// tricky when we are resuming form a previous match...
+	auto clear_grounding = [&](const PatternTermPtr& p)
```

Comment:
How about, say, after grounding it to (blob1=g1 blob2=g2 var=g3), it turns out that it's not a valid grounding because the var happens to exist in some other term as well and it's expected to be g4, so we will have to go back to glob_compare for the same candidate and try to ground var to something else (g4), would it also work if I save the glob groundings in the same stack where variable groundings?